src/api/user-recommendations/controllers/user-recommendations.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const NodeCache = require('node-cache');
+
+/**
+ * Read the documentation (https://strapi.io/documentation/developer-docs/latest/development/backend-customization.html#core-controllers)
+ * to customize this controller
+ */
+
+const myCache = new NodeCache({ stdTTL: 60 });
+
+module.exports = {
+  async recommend(ctx) {
+    let users = myCache.get('randomUsers');
+
+    // check if cache already has random users
+    if (users == undefined) {
+      const userCount = await strapi.entityService.count(
+        'plugin::users-permissions.user'
+      );
+
+      // gets 100 random user ids
+      const randomUserIds = Array(100)
+        .fill()
+        .map(() => Math.floor(Math.random() * userCount) + 1);
```

Comment:
How do we make sure there is no duplicate?

Reply:
I tested it on my local environment with 150 users in the database and usually ended up getting about 70 unique random ids. If there are duplicates, then less than 100 random users will be selected, but recommendations can still be made. I could change it so that there will always be 100 unique random ids, if that would be better.

Reply:
I'm okay with this. As we get more users, the probability of duplication gets lower.